plugins/node/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -0,0 +1,749 @@
+        );
+      }
+      return original.apply(this, arguments);
+    };
+  }
+
+  private getAckAllPatch(
+    isRejected: boolean,
+    endOperation: EndOperation,
+    original: Function
+  ) {
+    const self = this;
+    return function ackAll(
+      this: InstrumentationConsumeChannel,
+      requeueOrEmpty?: boolean
+    ): void {
+      self.endAllSpansOnChannel(this, isRejected, endOperation, requeueOrEmpty);
+      return original.apply(this, arguments);
+    };
+  }
+
+  private getAckPatch(
+    isRejected: boolean,
+    endOperation: EndOperation,
+    original: Function
+  ) {
+    const self = this;
+    return function ack(
+      this: InstrumentationConsumeChannel,
+      message: amqp.Message,
+      allUpToOrRequeue?: boolean,
+      requeue?: boolean
+    ): void {
+      const channel = this;
+      // we use this patch in reject function as well, but it has different signature
+      const requeueResolved =
+        endOperation === EndOperation.Reject ? allUpToOrRequeue : requeue;
+
+      const spansNotEnded: { msg: amqp.Message }[] =
+        channel[CHANNEL_SPANS_NOT_ENDED] ?? [];
+      const msgIndex = spansNotEnded.findIndex(
+        msgDetails => msgDetails.msg === message
+      );
+      if (msgIndex < 0) {
+        // should not happen in happy flow
+        // but possible if user is calling the api function ack twice with same message
+        self.endConsumerSpan(
+          message,
+          isRejected,
+          endOperation,
+          requeueResolved
+        );
+      } else if (endOperation !== EndOperation.Reject && allUpToOrRequeue) {
+        for (let i = 0; i <= msgIndex; i++) {
+          self.endConsumerSpan(
+            spansNotEnded[i].msg,
+            isRejected,
+            endOperation,
+            requeueResolved
+          );
+        }
+        spansNotEnded.splice(0, msgIndex + 1);
+      } else {
+        self.endConsumerSpan(
+          message,
+          isRejected,
+          endOperation,
+          requeueResolved
+        );
+        spansNotEnded.splice(msgIndex, 1);
+      }
+      return original.apply(this, arguments);
+    };
+  }
+
+  protected getConsumePatch(
+    moduleVersion: string | undefined,
+    original: Function
+  ) {
+    const self = this;
+    return function consume(
+      this: InstrumentationConsumeChannel,
+      queue: string,
+      onMessage: (msg: amqp.ConsumeMessage | null) => void,
+      options?: amqp.Options.Consume
+    ): Promise<amqp.Replies.Consume> {
+      const channel = this;
+      if (
+        !Object.prototype.hasOwnProperty.call(channel, CHANNEL_SPANS_NOT_ENDED)
+      ) {
+        if (self._config.consumeTimeoutMs) {
+          const timer = setInterval(() => {
+            self.checkConsumeTimeoutOnChannel(channel);
+          }, self._config.consumeTimeoutMs);
+          timer.unref();
+          Object.defineProperty(channel, CHANNEL_CONSUME_TIMEOUT_TIMER, {
+            value: timer,
+            enumerable: false,
```

Comment:
I'm not sure if it is really needed to use `Object.defineProperties` for symbol properties because they are never included in e.g. `Object.keys()` or in `for in` loops.
using `channel[CHANNEL_CONSUME_TIMEOUT_TIMER]` would be easier to read and faster